test/run_individual_tests.d

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env rdmd
+/**
+Allows running tests individually
+
+Usage:
+  ./run_individual_tests.d <test-file>...
+
+Example:
+  ./run_individual_tests.d runnable/template2962.d fail_compilation/fail282.d
+
+See the README.md for all available test targets
+*/
+
+void main(string[] args)
+{
+    import std.conv, std.format, std.getopt, std.path, std.process, std.stdio, std.string;
+    import std.parallelism : totalCPUs;
+
+    int jobs = totalCPUs;
+    auto res = getopt(args,
+        "j|jobs", "Specifies  the number of jobs (commands) to run simultaneously (default: %d)".format(totalCPUs), &jobs,
+    );
+    if (res.helpWanted || args.length < 2)
+    {
+        defaultGetoptPrinter(`./run_individual_tests.d <test-file>...
+
+Examples:
+
+    ./run_individual_tests.d runnable/template2962.d
+    ./run_individual_tests.d runnable/template2962.d fail_compilation/fail282.d
+
+See the README.md for a more in-depth explanation of the test-runner.
+`, res.options);
+        return;
+    }
+
+    string[] targets;
+    foreach (file; args[1 .. $])
+    {
+        auto normalizedDir = file.absolutePath.dirName;
+        targets ~= "test_results/%s/%s.out".format(normalizedDir.baseName, file.baseName);
+    }
+    auto scriptDir = __FILE_FULL_PATH__.dirName;
+    auto makeArguments = targets;
+    makeArguments ~= "--jobs=".text(jobs);
+    spawnProcess("make" ~ makeArguments, null, Config.none, scriptDir).wait;
```

Comment:
With #8129, all that is left to replace the Makefile is:
- only add targets which have a target with an older or non-existing timestamp
- replace jobs with `foreach (target; targets.parallel(1))`
- ensure the `test_results` directories exists
- set the environment flags
- resolve shorthands like `all` or `run_fail_compilation_tests`
- ensure that the test sanitizers are build
- add `if (targets.canFind("clean") { return removeRecursive(targetDir);}`